### PR TITLE
feat: add autocomplete support for locales

### PR DESCRIPTION
**Pull request by ST-DDT**
I will automate the generation/update for `KnownLocale`s in a later PR.

The is the TypeScript related part for https://github.com/faker-js/faker/issues/210

### CODE REVIEW

**Comment by ST-DDT** [comment]
This doesn't work yet due to `export = locales;` in `src/locales/index.ts`.

File: src/index.ts [hunk 1]
```diff
@@ -186,12 +190,13 @@ export interface LocaleDefinition {
   [group: string]: any;
 }
 
+export type UseableLocale = LiteralUnion<KnownLocale>;
```
Comment (Shinigami92) [inline src/index.ts:193]:
```suggestion
export type UsableLocale = LiteralUnion<KnownLocale>;
```

Comment (Shinigami92) [reply src/index.ts:193]: and all below...

**Review by Shinigami92 — CHANGES_REQUESTED** [review]

**Review by Shinigami92 — DISMISSED** [review]

**Review by Shinigami92 — APPROVED** [review]

**Review by prisis — APPROVED** [review]

File: src/locales/index.ts [hunk 2]
```diff
@@ -56,7 +56,67 @@ import zh_CN from './zh_CN';
 import zh_TW from './zh_TW';
 import zu_ZA from './zu_ZA';
 
-const locales: { [lang: string]: LocaleDefinition } = {
+export type KnownLocale =
```
[the remark this answers is not in the document]
Comment (pkuczynski) [reply src/locales/index.ts:59]: yes, first create the locales, and then below it, create the `KnownLocale` type. There is no circular reference if they stay in the same file.

[the remark this answers is not in the document]
Comment (pkuczynski) [reply src/locales/index.ts:59]: Something like this I had in mind:

```ts
const locales = {
  af_ZA,
  ar,
  ...
};

export type KnownLocale = keyof typeof locales

export default locales;
```

image

[the remark this answers is not in the document]
Comment (ST-DDT) [reply src/locales/index.ts:59]: Could you please create an issue (or PR) for that?
I currently work on something different and don't want to forget about it.

**Pull request merged**